pilot/pkg/model/sidecar.go

```diff
@@ -78,6 +78,10 @@ type SidecarScope struct {
 	// destination rule.
 	destinationRules map[config.Hostname]*Config
 
+	// A given hostname should only be considered in a single namespace. This mapping determines which
+	// namespace a hostname exists in
+	NamespaceForHostname map[config.Hostname]string
+
```

Comment:
Do you need this ? Sidecar is already a filtered list. Why don’t you filter when initializing the sidecar?